pymodbus/server/async_io.py

```diff
@@ -173,17 +175,28 @@ def execute(self, request, *addr):
         if self.server.request_tracer:
             self.server.request_tracer(request, *addr)
 
+        task = self.loop.create_task(self._async_execute(request, *addr))
+        self.request_tasks.add(task)
+        task.add_done_callback(self.request_tasks.discard)
+
+    async def _async_execute(self, request, *addr):
         broadcast = False
+        response = None
         try:
             if self.server.broadcast_enable and not request.slave_id:
                 broadcast = True
                 # if broadcasting then execute on all slave contexts,
                 # note response will be ignored
                 for slave_id in self.server.context.slaves():
                     response = request.execute(self.server.context[slave_id])
+                    if asyncio.iscoroutine(response):
```

Comment:
A short comment telling this is temporary would be nice.

Reply:
I'll add mention about temporary